webhook/resources/backup/validator.go

```diff
@@ -92,3 +98,41 @@ func (b *backupValidator) Create(request *admission.Request, newObj runtime.Obje
 
 	return nil
 }
+
+func (b *backupValidator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) error {
+	oldBackup, ok := oldObj.(*longhorn.Backup)
+	if !ok {
+		return werror.NewInvalidError(fmt.Sprintf("%v is not a *longhorn.Backup", oldObj), "")
+	}
+	newBackup, ok := newObj.(*longhorn.Backup)
+	if !ok {
+		return werror.NewInvalidError(fmt.Sprintf("%v is not a *longhorn.Backup", newObj), "")
+	}
+
+	// Allow backup block size mutation only when the existing obj is not set, or correcting the existed invalid value
+	isValidOldBackupBlockSize := b.validateBackupBlockSize(oldBackup, true) == nil
+	if isValidOldBackupBlockSize && oldBackup.Spec.BackupBlockSize != newBackup.Spec.BackupBlockSize {
+		err := fmt.Errorf("changing backup block size for backup %v is not supported", oldBackup.Name)
+		return werror.NewInvalidError(err.Error(), "")
+	}
```

Comment:
If `oldBackup.Spec.BackupBlockSize` is `-1` and `newBackup.Spec.BackupBlockSize` is `2MiB`, the `isValidOldBackupBlockSize` will be true, but the request will be rejected. Isn't it? Then, user cannot correct the value.